### PR TITLE
4445 bbmap pytest to nf test

### PR DESCRIPTION
**Pull request by tfalkarkea**
## PR checklist

Contributes to https://github.com/nf-core/modules/issues/4445, but only one subtool for now (bbduk). 

Pytest removed and ported to nf-test.

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If necessary, include test data in your PR (uses existing test data)
- [x] Follow the naming conventions.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`

### CODE REVIEW

**Review by johnoooh — APPROVED** [review]

**Review by ggabernet — APPROVED** [review]
Looks good to me!

File: modules/nf-core/bbmap/bbduk/environment.yml [hunk 1]
```diff
@@ -4,4 +4,4 @@ channels:
   - bioconda
   - defaults
 dependencies:
-  - bioconda::bbmap=39.01
+  - bioconda::bbmap=38.18
```
Comment (apetkau) [inline modules/nf-core/bbmap/bbduk/environment.yml:7]: This is moving back a version for conda, but the singularity/docker versions is still `39.01`? Should these all be made consistent? Was this to get tests to work with conda?

**Review by apetkau — COMMENTED** [review]
This looks awesome. I love how you added so many tests 😄 . I just had one comment.

File: modules/nf-core/bbmap/bbduk/environment.yml (hunk 1 above)
Comment (tfalkarkea) [reply modules/nf-core/bbmap/bbduk/environment.yml:7]: This explains why their md5s are clashing! Yes let me move back the others.

**Comment by sateeshperi** [comment]
did u re-generate snapshot after changing the version ?

**Comment by tfalkarkea** [comment]
> did u re-generate snapshot after changing the version ?

Yup, and I'm seeing the same problems while running conda and docker on my machine.

**Pull request merged**